File: n2/2.py
cups = list(map(int, input("Введите кол-во полных банок и эквивалент обмена: ").split())) # [полные, обмен]
drunk = cups[0] 
i = 0
full = 0
if cups[0] > 0:
    while cups[0] >= cups[1]:
        i += 1 # шаг опустошения
        full = cups[0] // cups[1] # обмен на полную банку
        i += 1 # шаг обмена
        empty = cups[0] % cups[1] # кол-во пустых банок
        drunk += full # кол-во выпитых
        cups[0] = full + empty
    i += 1 # выпиваем оставшиеся     
    print(drunk, i)

File: n2/2.cs
using System;
class Program {
    static void Main() {
        Console.Write("Введите кол-во полных банок и эквивалент обмена: ");
        string[] input = Console.ReadLine().Split();
        int[] cups = { int.Parse(input[0]), int.Parse(input[1]) };

        int drunk = cups[0]; // изначально выпито
        int i = 0;           // счётчик шагов
        int full = 0;        // результат обмена
        if (cups[0] < 1)  return;
        // пока можно обменивать
        while (cups[0] >= cups[1]) {
            i += 1; // опустошаем
            full = cups[0] / cups[1]; // обмен
            i += 1;
            int empty = cups[0] % cups[1]; // остаток
            drunk += full;
            cups[0] = full + empty; // обновление
        }
        i += 1; // допиваем
        Console.WriteLine("Выпито банок и кол-во итераций: " + drunk + " " + i);
    }
}
